huject/index.d.ts

```diff
@@ -0,0 +1,273 @@
+
+    /**
+     * Resolve instance by string definition and pass to constructor argument
+     * @param literal string definition
+     * @param method Optional factory method
+     * @example
+     * @ConstructorInject
+     * Class MyClass {
+     *     private service: MyService;
+     *     private anotherService: AnotherService;
+     *
+     *     private tokenStr: string;
+     *
+     *     public constructor(
+     *         service: MyService,
+     *         @ConstructorInject('service', FactoryMethod.SINGLETON) anotherService: AnotherService,
+     *         @ConstructorInject('serviceToken') token: string
+     *     ) {
+     *         this.service = service;
+     *         this.anotherService = anotherService;
+     *         this.tokenStr = token;
+     *     }
+     * }
+     */
+    export function ConstructorInject(literal: string, method?: FactoryMethod): void;
+
+    /**
+     * Specify that property injection should be optional. That means if dependency couldn't be resolved, then leave original specified value
+     * @param target
+     * @param propertyKey
+     * @example
+     * class MyClass {
+     *     @Optional
+     *     @Inject('coolnumber')
+     *     public num: number = 50;
+     *
+     *     @Optional
+     *     @Inject('classToken')
+     *     public token: string = 'defaultToken';
+     * }
+     */
+    export function Optional(target: Object, propertyKey: string|symbol): void;
+
+    /**
+     * Specify that constructor argument injection should be optional. If argument couldn't be resolved, pass null instead
+     * @param target
+     * @param propertyKey
+     * @param parameterIndex
+     * @example
+     * @ConstructorInject
+     * Class MyClass {
+     *     private service: MyService;
+     *     public constructor(@Optional @ConstructorInject(FactoryMethod.SINGLETON) service: MyService) {
+     *         if (service) {
+     *             this.service = service;
+     *         }
+     *     }
+     * }
+     */
+    export function Optional(target: Object, propertyKey: string|symbol, parameterIndex: number): void;
+
+    /**
+     * Specify that class should be base for auto-factory
+     * @param target
+     * @example
+     * @Factory
+     * class MyModelFactory {
+     *     @Factory
+     *     public createModel(): Model {return null;}
+     * }
+     */
+    export function Factory<TFunction extends Function>(target: TFunction): TFunction | void;
+
+    /**
+     * Specify that method should be used for auto-factory
+     * @param target
+     * @param propertyKey
+     * @param descriptor
+     * @example
+     * @Factory
+     * class MyModelFactory {
+     *     @Factory
+     *     public createModel(): Model {return null;}
+     * }
+     */
+    export function Factory<T>(target: Object, propertyKey: string|symbol, descriptor: TypedPropertyDescriptor<T>): TypedPropertyDescriptor<T> | void;
+}
+
+declare module "huject" {
```

Comment:
could you strip `declare module "huject" {`?
https://github.com/Microsoft/types-publisher#fields-in-datadefinitionsjson
> ProperModule: Only top-level import and export declarations. Preferred